lib/cocoapods/generator/copy_xcframework_script.rb

```diff
@@ -93,7 +92,18 @@ def script
     target_variant="maccatalyst"
   fi
   for i in ${!paths[@]}; do
-    if [[ "${paths[$i]}" == *"$target_arch"* ]] && [[ "${paths[$i]}" == *"$target_variant"* ]]; then
+    local matched_all_archs="1"
+    for target_arch in $target_archs
+    do
+      if ! [[ "${paths[$i]}" == *"$target_variant"* ]]; then
+        matched_all_archs="0"
+      fi
+      if ! [[ "${paths[$i]}" == *"$target_arch"* ]]; then
```

Comment:
since we are checking each architecture individually, could we remove the `*` and check that they are exactly equal? I think that would resolve the `armv7` case

Reply:
ah nevermind I see why that woulnd't work now, we're still checking the whole list against the target

Reply:
We could check if we match one of `-$target_arch_`, `_$target_arch_` or `_$target_arch-`, then it's a proper check. But still quite nasty.

Reply:
maybe we could use `grep` with a regex? I don't mean to delay this improvement but I also don't want to swap one bug for another 😅

Reply:
its possible a framework _only_ supports 1 arch, and therefore wouldn't have `_` as a prefix & suffix

Reply:
Then we also need to add -$target_arch- to the list. It't not really swapping bugs, this problem already exists in the current implementation. I will check whether we can fix it with a regex :)

Reply:
@amorde I updated the PR to use a regex. I think this is a solid solution for now :)

Reply:
It's not completely working, for some reason x86_64 doesn't work in the regex

Reply:
Fixed! :)